src/Xamarin.Android.Build.Tasks/Tasks/CheckForInvalidResourceFileNames.cs

```diff
@@ -17,13 +17,22 @@ public class CheckForInvalidResourceFileNames : Task {
 		public ITaskItem[] Resources { get; set; }
 
 		Regex fileNameCheck = new Regex ("[^a-zA-Z0-9_.]+", RegexOptions.Compiled);
+		Regex fileNameWithHyphenCheck = new Regex ("[^a-zA-Z0-9_.-]+", RegexOptions.Compiled);
 
 		public override bool Execute ()
 		{
 			foreach (var resource in Resources) {
-				var match = fileNameCheck.Match (Path.GetFileName (resource.ItemSpec));
-				if (match.Success) {
-					Log.LogCodedError ("APT0000", resource.ItemSpec, 0, "Invalid file name: It must contain only [a-zA-Z0-9_.].");
+				var fileName = Path.GetFileName (resource.ItemSpec);
+				if (string.Compare (Directory.GetParent (resource.ItemSpec).Name, "values", StringComparison.OrdinalIgnoreCase) == 0) {
+					var match = fileNameWithHyphenCheck.Match (fileName);
+					if (match.Success) {
+						Log.LogCodedError ("APT0000", resource.ItemSpec, 0, "Invalid file name: It must contain only [a-zA-Z0-9_.-].");
```

Comment:
`Regex.ToString()` prints the pattern provided to the constructor, so instead of "duplicating" the pattern, we should use that:

```
Log.LogCodedError ("APT0000", resource.ItemSpec, 0, $"Invalid file name: It must contain only `{fileNameWithHyphenCheck}`.");
```